generic/piranha/test-resources/kt/feature_flag_system_2/treated/configurations/rules.toml

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2022 Uber Technologies, Inc.
+# 
+# <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of the License at
+# <p>http://www.apache.org/licenses/LICENSE-2.0
+# 
+# <p>Unless required by applicable law or agreed to in writing, software distributed under the
+# License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file contains rules to the specific feature flag API.
+# Before :
+# internal interface SomeParameter {
+#     @BoolParam(key = "STALE_FLAG", namespace = "some_long_name")
+#     fun isStaleFeature(): BoolParameter
+#     @BoolParam(key = "other_flag", namespace = "some_long_name")
+#     fun isOtherFlag(): BoolParameter
+# }
+# After :
+# internal interface SomeParameter {
+#     @BoolParam(key = "other_flag", namespace = "some_long_name")
+#     fun isOtherFlag(): BoolParameter
+# }
+#
+[[rules]]
+name = "delete_flag_method_declaration_fully_annotated"
+query = """
+(
+(function_declaration 
+  (modifiers (annotation (constructor_invocation (_)@annotation_name (value_arguments (value_argument (_)@lhs (_)@rhs))@args )  ) )
+  ((simple_identifier) @m_name) 
+) @function_declaration
+(#eq? @lhs "key")
+(#eq? @rhs "\\\"@stale_flag_name\\\"")
+)
+(
+(function_declaration 
+  (modifiers (annotation (constructor_invocation (_) (value_arguments (value_argument (_)@lhs (_)@rhs))@args ) )) 
+) @pd2
+(#eq? @lhs "namespace")
+(#eq? @annotation_name "BoolParam")
+(#eq? @rhs "\\\"@namespace\\\"")
+)
+"""
+replace_node = "function_declaration"
+replace = ""
+groups = ["delete_property_declaration"]
+holes = ["stale_flag_name", "namespace"]
+
+# Before :
+# @ParameterDefinition(namespace = "some_long_name")
+# internal interface SomeOtherParameter {
+#     @BoolParam(key = "STALE_FLAG")
+#     fun isStaleFeature(): BoolParameter
+#     @BoolParam(key = "other_flag", namespace = "some_long_name")
+#     fun isOtherFlag(): BoolParameter
+# }
+# After :
+# @ParameterDefinition(namespace = "some_long_name")
+# internal interface SomeOtherParameter {
+#     @BoolParam(key = "other_flag", namespace = "some_long_name")
+#     fun isOtherFlag(): BoolParameter
+# }
+#
+[[rules]]
+name = "delete_flag_method_declaration_inside_annotated_interface"
```

Comment:
defining these rules is going to be really difficult for anyone not familiar with TreeSitter.

Reply:
haha! Yea. It requires getting used to reading, comprehending and creating `s-expressions`.